lightning/src/ln/channelmanager.rs

```diff
@@ -10645,14 +10698,22 @@ where
 				}
 			},
 			#[cfg(async_payments)]
-			OffersMessage::StaticInvoice(_invoice) => {
-				match responder {
-					Some(responder) => {
-						responder.respond(OffersMessage::InvoiceError(
-								InvoiceError::from_string("Static invoices not yet supported".to_string())
-						))
-					},
+			OffersMessage::StaticInvoice(invoice) => {
+				let responder = match responder {
+					Some(responder) => responder,
 					None => return ResponseInstruction::NoResponse,
+				};
+				let payment_id = match context {
+					OffersContext::OutboundPayment { payment_id } => payment_id,
+					_ => {
+						return responder.respond(OffersMessage::InvoiceError(
+								InvoiceError::from_string("Unrecognized invoice".to_string())
+						))
```

Comment:
Let's not respond if we the reply path isn't being used in the right context. I made a similar change in #3139.